### PR TITLE
proper nextjs docker configs, reduce image size

### PR DESCRIPTION
**Pull request by Lxkas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
 		"react": "18.2.0",
 		"react-dom": "18.2.0",
 		"react-icons": "^4.11.0",
+		"sharp": "^0.32.6",
```
Comment (Lxkas) [inline package.json:34]: this is required for default image optimization loader.
https://nextjs.org/docs/pages/api-reference/next-config-js/output#automatically-copying-traced-files

File: dockerfile [hunk 2]
```diff
@@ -1,26 +1,56 @@
-# ---- Base Node ----
-FROM node:alpine AS base
+FROM node:18-alpine AS base
```
Comment (Lxkas) [inline dockerfile:1]: This locks us in to node 18, but honestly.. we probably won't need any cutting edge features as of yet.
Add to todo ;P

File: dockerfile [hunk 3]
```diff
@@ -1,26 +1,56 @@
...
+
+# Install dependencies only when needed
+FROM base AS deps
+
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
```
Comment (chinathaip) [inline dockerfile:6]: cool..

Comment (Lxkas) [reply dockerfile:6]: Cool indeed.
( I still have no idea )

**Review by chinathaip — APPROVED** [review]
one of the longest dockerfile I've reviewd lmao

File: dockerfile (hunk 3 above)
Comment (chinathaip) [reply dockerfile:6]: (me too)

File: dockerfile (hunk 2 above)
Comment (chinathaip) [reply dockerfile:1]: does the latest alpine not work?

**Pull request merged**